src/Serilog.Sinks.EventLog/Sinks/EventLog/EventLogSink.cs

```diff
@@ -53,69 +58,153 @@ public EventLogSink(string source, string logName, ITextFormatter textFormatter,
-		/// <remarks>
-		/// <see cref="LogEventLevel.Debug" />, <see cref="LogEventLevel.Information" /> and <see cref="LogEventLevel.Verbose" /> are registered as <see cref="EventLogEntryType.Information" />.
-		/// <see cref="LogEventLevel.Error" />, <see cref="LogEventLevel.Fatal" /> are registered as <see cref="EventLogEntryType.Error" />.
-		/// <see cref="LogEventLevel.Warning" /> are registered as <see cref="EventLogEntryType.Warning" />.
-		/// The Event ID in the Windows log will be set to the integer value of the <paramref name="logEvent"/>'s <see cref="LogEvent.Level"/> property, so that the log can be filtered with more granularity.</remarks>
-		public void Emit(LogEvent logEvent)
-		{
-			if (logEvent == null) throw new ArgumentNullException("logEvent");
-
-			EventLogEntryType type;
-			switch (logEvent.Level)
-			{
-				case LogEventLevel.Debug:
-				case LogEventLevel.Information:
-				case LogEventLevel.Verbose:
-					type = EventLogEntryType.Information;
-					break;
-
-				case LogEventLevel.Error:
-				case LogEventLevel.Fatal:
-					type = EventLogEntryType.Error;
-					break;
-
-				case LogEventLevel.Warning:
-					type = EventLogEntryType.Warning;
-					break;
-
-				default:
-					SelfLog.WriteLine("Unexpected logging level, writing to EventLog as Information");
-					type = EventLogEntryType.Information;
-					break;
-			}
-
-			var payloadWriter = new StringWriter();
-			_textFormatter.Format(logEvent, payloadWriter);
-
-
-            //The payload is limitted in length and allowed chars
-            //see: https://msdn.microsoft.com/en-us/library/e29k5ebc%28v=vs.110%29.aspx
-            var payload = payloadWriter.ToString();
+        /// Returns name of EventLog that contains the registered source
+        /// </summary>
+        /// <param name="source"></param>
+        /// <param name="machineName"></param>
+        /// <returns></returns>
+	    private System.Diagnostics.EventLog GetLogFromSource(string source, string machineName)
+	    {
+	        return System.Diagnostics.EventLog.GetEventLogs().ToList().FirstOrDefault(l => SourceExistsInLog(source, l.Log, machineName));
+	    }
+
+	    private bool SourceExistsInLog(string sourceName, string logName, string machineName)
+	    {
+            return FindSourceNamesFromLog(logName, machineName)?.Any(s => s == sourceName) ?? false;
+	    }
+
+        /// <summary>
+        /// Returns list of sources associated with specific log
+        /// </summary>
+        /// <remarks>
+        /// EventLog Api doesn't support this natively so we have to use the Registry
+        /// </remarks>
+        private static List<string> FindSourceNamesFromLog(string logName, string machineName)
+        {
+            if (machineName.Trim() == ".") machineName = Environment.MachineName;
+
+            var hive = RegistryKey.OpenRemoteBaseKey(RegistryHive.LocalMachine, machineName);
+            var keyLog = hive.OpenSubKey($@"SYSTEM\CurrentControlSet\Services\Eventlog\{logName}");
```

Comment:
Bah, disregard; this code is garbage.  I missed an available api in the `EventLog` class - `LogNameFromSourceName` - which I am rewriting this logic to make use of.

Also, to do anything admistrative in the event log (e.g. create an event source/log), you need admin privilages (see remarks [here](https://msdn.microsoft.com/en-us/library/5zbwd3s3(v=vs.110).aspx)) from MSDN:

> To create an event source in Windows Vista and later or Windows Server 2003, you must have administrative privileges.
The reason for this requirement is that all event logs, including security, must be searched to determine whether the event source is unique. Starting with Windows Vista, users do not have permission to access the security log; therefore, a SecurityException is thrown.

Sooo... TL;DR I don't believe we're any worse off permissions-requireme-wise than using (in the last sink version) a new source in the `Application` log.